tests/chainer_tests/links_tests/connection_tests/test_convolution_2d.py

```diff
@@ -151,74 +111,73 @@ def test_col2im_consistency(self):
         im_gpu = conv.col2im_gpu(cuda.to_gpu(col), 2, 2, 1, 1, h, w)
         testing.assert_allclose(im_cpu, im_gpu.get())
 
-    def check_forward_consistency(self):
-        x_cpu = chainer.Variable(self.x)
-        y_cpu = self.link(x_cpu)
-        self.assertEqual(y_cpu.data.dtype, numpy.float32)
 
-        self.link.to_gpu()
-        x_gpu = chainer.Variable(cuda.to_gpu(self.x))
-        y_gpu = self.link(x_gpu)
-        self.assertEqual(y_gpu.data.dtype, numpy.float32)
-
-        testing.assert_allclose(y_cpu.data, y_gpu.data.get())
+@testing.parameterize(*testing.product({
+    'conv_args': [((None, 2, 3, 2, 1), {}),
+                  ((2, 3), {'stride': 2, 'pad': 1})],
+}))
+@testing.inject_backend_tests(
+    None,
+    # CPU tests
+    [{}]
+    # GPU tests
+    + testing.product({
+        'use_cuda': [True],
+        'use_cudnn': ['never', 'always'],
+        'cuda_device': [0, 1],
+    })
+    + [
+        {'use_chainerx': True, 'chainerx_device': 'native:0'},
+        {'use_chainerx': True, 'chainerx_device': 'cuda:0'},
+        {'use_chainerx': True, 'chainerx_device': 'cuda:1'},
+    ])
+class TestConvolution2DParameterShapePlaceholder(testing.LinkTestCase):
```

Comment:
How about merging this test with `TestConvolution2D`? (not mandatory)
Is there any reason this test exists separately?

Reply:
I am quite unsure of it, the only difference I see is that it explicitly copies the link parameters after initialization,  besides that they look identical ...